src/main/java/uk/gov/ch/developer/docs/security/WebSecurity.java

```diff
@@ -5,25 +5,39 @@
 import org.springframework.security.config.annotation.web.builders.HttpSecurity;
 import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
 import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.web.authentication.www.BasicAuthenticationFilter;
+import uk.gov.companieshouse.session.handler.SessionHandler;
 
 @EnableWebSecurity
-public class WebSecurity extends WebSecurityConfigurerAdapter {
+public class WebSecurity {
 
     @Configuration
     @Order(1)
-    public static class TemporaryStartPageSecurityConfig extends WebSecurityConfigurerAdapter {
+    public static class RootSecurityConfig extends WebSecurityConfigurerAdapter {
         @Override
         protected void configure(HttpSecurity http) throws Exception {
-            http.antMatcher("/signin");
+            http.antMatcher("/").addFilterBefore(new SessionHandler(),
+                    BasicAuthenticationFilter.class);
         }
     }
-
     @Configuration
     @Order(2)
-    public static class DocsWebSecurityFilterConfig extends WebSecurityConfigurerAdapter {
+    public static class SignInSecurityConfig extends WebSecurityConfigurerAdapter {
+        @Override
+        protected void configure(HttpSecurity http) throws Exception {
+            http.antMatcher("/signin").addFilterBefore(new SessionHandler(),
+                    BasicAuthenticationFilter.class);
+        }
+    }
+
+    @Configuration
+    @Order(3)
+    public static class CallbackSecurityConfig extends WebSecurityConfigurerAdapter {
         @Override
         protected void configure(HttpSecurity http) throws Exception {
-            http.antMatcher("/**");
+            http.antMatcher("/oauth2/user/**").addFilterBefore(new SessionHandler(),
+                    BasicAuthenticationFilter.class);
         }
```

Comment:
We should probably combine these into one `@Configuration` as we have the same handler for all of the matched paths.

Reply:
Please post words